projects/cdk/directives/font-size/font-size.directive.ts

```diff
@@ -0,0 +1,64 @@
+import {DOCUMENT, isPlatformBrowser} from '@angular/common';
+import {DestroyRef, Directive, inject, type OnInit, PLATFORM_ID} from '@angular/core';
+import type {TuiStringHandler} from '@taiga-ui/cdk/types';
+import {tuiCreateTokenFromFactory, tuiInjectElement} from '@taiga-ui/cdk/utils';
+
+const IFRAME = 'position: fixed; visibility: hidden; pointer-events: none';
+const BODY = 'height: fit-content; line-height: 1em;';
+
+export const TUI_FONT_SIZE_HANDLER = tuiCreateTokenFromFactory<TuiStringHandler<number>>(
+    () => (size) => `${Math.max(size, 16)}`,
+);
+
+@Directive({
+    standalone: true,
+    host: {
+        '(window:resize.zoneless)': 'onResize()',
+    },
+})
+export class TuiFontSize implements OnInit {
+    private readonly el = tuiInjectElement();
+    private readonly doc = inject(DOCUMENT);
+    private readonly iframe = this.doc.createElement('iframe');
+    private readonly handler = inject(TUI_FONT_SIZE_HANDLER);
+    private readonly isBrowser = isPlatformBrowser(inject(PLATFORM_ID));
+    private readonly destroyRef = inject(DestroyRef);
+
+    public ngOnInit(): void {
+        this.doc.body.append(this.iframe);
+        this.iframe.setAttribute('style', IFRAME);
+        this.onResize();
+
+        if (!this.isBrowser) {
+            return;
+        }
+
+        const doc = this.iframe.contentWindow?.document;
+        const observer = new ResizeObserver(() =>
+            this.el.style.setProperty(
+                '--tui-font-size',
+                this.handler(doc?.body.offsetHeight || 0),
+            ),
+        );
+
+        doc?.documentElement.style.setProperty('font', '-apple-system-body');
+        doc?.body.setAttribute('style', BODY);
+        doc?.body.insertAdjacentText('beforeend', '.'.repeat(1000));
```

Comment:
Chrome on Android has a bug where it does not react to system font size scaling:
https://issues.chromium.org/issues/375209519
However empirically I discovered that it starts reacting once enough text is accumulated on the page, 100 dots is not enough, 1000 dots seems to be enough for all screen sizes.